test/Internal/TypeChecking/Positivity/Occurrence.hs

```diff
@@ -136,13 +136,12 @@ prop_boundToEverySome1 (NonEmpty w) =
 prop_boundToEverySome2 :: Property
 prop_boundToEverySome2 =
   forAll (elements $ Map.toList boundToEverySome) $ \(bound, ess) ->
-    (forAll (oneof [ do os1 <- listOf (arbitrary `suchThat` every)
-                        o   <- arbitrary
-                                 `suchThat` (\o -> every o && some o)
-                        os2 <- listOf (arbitrary `suchThat` every)
-                        return (os1 ++ [o] ++ os2)
-                   | (every, some) <- ess
-                   ]) $ \w ->
+    forAll (oneof [ do os1 <- listOf (arbitrary `suchThat` every)
+                       o   <- arbitrary `suchThat` (\o -> every o && some o)
+                       os2 <- listOf (arbitrary `suchThat` every)
+                       return (os1 ++ [o] ++ os2)
+                  | (every, some) <- ess
+                  ]) (\w ->
```

Comment:
I don't like this change.  The structure is more clear without it.

Reply:
To explain:  In the old version the top-level conjuction is more clearly visible as `(...) .&&. (...)`.

One can raise objects against writing such monster expressions in general.  I think the best would be to introduce some named subformulas, e.g. for the two conjuncts.

Reply:
I did this and for lack of better names went with `left .&&. right`. I didn't follow the next round of HLint suggestions, leaving that for another time.

```
$ hlint test/Internal/TypeChecking/Positivity/Occurrence.hs
test/Internal/TypeChecking/Positivity/Occurrence.hs:110:5-19: Suggestion: Use null
Found:
  length ess >= 1
Perhaps:
  not (null ess)
Note: increases laziness

test/Internal/TypeChecking/Positivity/Occurrence.hs:157:19-62: Warning: Redundant lambda
Found:
  every = \ o -> and [not (s o) | s <- ss]
Perhaps:
  every o = and [not (s o) | s <- ss]

test/Internal/TypeChecking/Positivity/Occurrence.hs:158:19-55: Warning: Redundant lambda
Found:
  some e = \ o -> every o && not (e o)
Perhaps:
  some e o = every o && not (e o)

test/Internal/TypeChecking/Positivity/Occurrence.hs:166:20-49: Warning: Use >
Found:
  not (foldr1 otimes w <= bound)
Perhaps:
  foldr1 otimes w > bound
Note: incorrect if either value is NaN

4 hints
```